@here/harp-mapview/lib/MapView.ts

```diff
@@ -2713,6 +2722,36 @@ export class MapView extends THREE.EventDispatcher {
             this.enqueueTextRendererCreation();
         }
 
+        // Check if this is the time to place the labels for the first time. Pretty much everything
+        // should have been loaded, and no animation should be running.
+        if (
+            !this.m_initialTextPlacementDone &&
+            !this.m_firstFrameComplete &&
+            !this.isDynamicFrame &&
+            !this.m_initialTextPlacementDone &&
+            !this.m_themeIsLoading &&
+            this.m_visibleTiles.allVisibleTilesLoaded &&
+            this.m_connectedDataSources.size + this.m_failedDataSources.size ===
+                this.m_tileDataSources.length &&
+            this.m_poiTableManager.finishedLoading
+        ) {
+            const textElementRendererFinished =
+                this.m_textElementsRenderer !== undefined &&
+                this.m_textElementsRenderer.ready &&
```

Comment:
Maybe instead of juggling with partial states of TER just expose proper `TER.ready` which checks also `loading` state.

Reply:
I pondered that idea as well. But the ready state is final, while the loading state may change when new character sets are being loaded, no?